### DISCUSSION TITLE
Duplicated test collection when importing

### DISCUSSION THREAD

**Discussion by joaodaher** [pytest-dev/pytest#11949] [Q&A] [answered]
When I import a test in another file, pytest duplicates the test case.
Here's a minimal example:

`/test_a.py`
```
def test_case_a():
    print("Test A :: Case A")
```

`/test_b.py`
```
from test_a import test_case_a

def test_case_b():
    print("Test B :: Case B")
```

When I run `pytest --co` I get:
```
platform darwin -- Python 3.12.0, pytest-8.0.0, pluggy-1.4.0
rootdir: [suppressed]/pytest-dups
collected 3 items

<Dir pytest-dups>
  <Module test_a.py>
    <Function test_case_a>
  <Module test_b.py>
    <Function test_case_a>
    <Function test_case_b>
```

As you can see, `test_case_a` is collected twice (probably due to being present on test_b's namespace?).

This can get a lot worse with class-based tests, since all of them are duplicated.

Is this a feature or a bug?
Is there a way of preventing tests from being imported in the "wrong" module?

**Comment by nicoddemus** [comment]
That's expected, pytest will list the contents of a module to find the tests; even if it was imported from another module, pytest will not do any checks regarding that.

I would say it is an unknown feature.

> Is there a way of preventing tests from being imported in the "wrong" module?

I don't think so, not for test functions.

**Reply by joaodaher** [reply]
Good to know, thanks for the reply.
I've opened an issue to investigate it further, maybe it's really a bug that I could contribute.

**Reply by RonnyPfannschmidt** [reply]
It's intentionally left in

**Reply by joaodaher** [reply]
@RonnyPfannschmidt , could you please explain more about this feature and its use cases?
Why importing a test would re-collect it to run it again?

**Reply by RonnyPfannschmidt** [reply, marked as answer]
This allows to import whole suite's of Tests into a file with different fixtures

**Reply by joaodaher** [reply]
Ok, I guess. Thanks for the help.
